activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb

```diff
@@ -453,6 +453,9 @@ def checkout_and_verify(c)
           c.verify!
         end
         c
+      rescue
+        disconnect!
```

Comment:
OK, so I made pull request to change to release only failed connection.
https://github.com/rails/rails/pull/18478